### ISSUE TITLE
Creation of duplicated entries

### ISSUE THREAD

**Issue by khillion** [bio-tools/biotoolsRegistry#275]
Hello,
I have been registering a new tool today and I have observed a weird behavior. Two entries have been created: https://bio.tools/Salmonella_CRISPR_typing and https://bio.tools/Salmonella_CRISPR_typing9706.
The only difference between these two entries being the fact that permissions have been granted to my account.
Have you ever observed this too ?
Thanks !

**Comment by hansioan** [comment]
Did you do it via the API or via the interface?

**Comment by khillion** [comment]
We did everything via the interface directly.

**Comment by hansioan** [comment]
I assume that you registered the tool initially and then updated it, and not register the tool twice with the same name. Could you reproduce this on the https://dev.bio.tools?

**Comment by hansioan** [comment]
@khillion I've made a dummy tool on dev:
https://dev.bio.tools/duplicate_tool
and granted you permission to edit. could you try and see if this happens?

**Comment by hansioan** [comment]
The number at the end of the link https://bio.tools/Salmonella_CRISPR_typing9706 should only happen if the tool with the exact same name is registered twice. The tool name will generate the tool id and if that same id already exists, a random number will be appended to the end. So, this should only occur if the tool is registered twice, and not if it is registered once and then updated, but of course there could be bugs.

**Comment by khillion** [comment]
Thanks @hansioan . I just tried to do the same as we did yesterday and I know what happened!
I created this entry: https://dev.bio.tools/duplicated_test and realized I forgot to share the permissions, so without exiting the registration interface, I went on permissions and added your account. I clicked `Save` and instead of editing the entry I just registered, it creates an other one instead: https://dev.bio.tools/duplicated_test7422.
So basically, I think that when you decide to register a new tool, as long as you stay on the registration interface and you click the `Save` button, it will create a new entry (I am not sure to be clear enough).
If I try to illustrate with an example:
* Click add content
* Fill up the minimum information required
* Click `Save`
* Add Permissions
* Click `Save`
I think this is going to create to entries.

**Comment by hansioan** [comment]
@khillion Yes, you are right! 
I created a different tool and filled in the info, saved and then added people who can edit the tool and saved again and the two saves generated the 2 tools. In fact, I think every time one adds new info and clicks save it creates a new entry....
@ekry @baileqi we need to fix this since it is super important that one can save often so they don't mistakenly close the window and lose all the info, but it should definitely not trigger the creation of a new tool every time.

**Comment by joncison** [comment]
@ekry @baileqi - giving this a nudge, super-important it's fixed ASAP

**Comment by joncison** [comment]
Both @hansioan and I confirm this is still broken (in dev.bio.tools):  saving more than once creates a new entry, each time the "Save" button is pressed!  This is obviously incredibly bad.

The behaviour of the "Save" button needs to change so that the first press does a "POST" and subsequent presses do a "PUT" (alternatively redirect you to the "edit page" for that tool).

cc @piotrgithub1

**Comment by joncison** [comment]
cc @khillion
